### PR TITLE
Converting policy page into a policy form inside the sign up page

### PR DESCRIPTION
**Pull request by beasrodrigues24**
Why:
* For it to be possible to use a button to return to a previous step, the 4 steps must be in the same page

How:
* By moving PolicyPage files to components/Forms/PolicyForm
* By creating a new step on the sign up and adapting across the code to fit this

### CODE REVIEW

File: frontend/src/pages/SignupPage/index.js [hunk 1]
```diff
@@ -141,9 +175,15 @@ const SignupPage = () => {
         <Logo />
       </div>
       <div className={styles.main}>{stepInfo[state.step].component}</div>
-      <Button onClick={continueToNextStep}>
-        {stepInfo[state.step].buttonText}
-      </Button>
+      {(() => {
+        if (state.step !== STEP_4) {
```
Comment (beasrodrigues24) [inline frontend/src/pages/SignupPage/index.js:179]: This if exists because the button on the step 4 will be styled differently with the new design

Comment (beasrodrigues24) [reply frontend/src/pages/SignupPage/index.js:179]: So it will come from inside the PrivacyForm component

**Comment by beasrodrigues24** [comment]
Also the Privacy page will be a little crooked, it will have to be fixed in another PR after we decide if we are keeping the Logo on top of the page or not

File: frontend/src/pages/SignupPage/index.js (hunk 1 above)
Comment (mirhamasala) [reply frontend/src/pages/SignupPage/index.js:179]: Why the function?

**Comment by mirhamasala** [comment]
Can you please check the CSS for this form? It overlaps with the SignUp page. Especially, `.root` and `.main` You might not need that.

File: frontend/src/pages/SignupPage/index.js (hunk 1 above)
Comment (mirhamasala) [reply frontend/src/pages/SignupPage/index.js:179]: Is this what you're looking for?

```
      {state.step !== STEP_4 && (
        <Button onClick={continueToNextStep}>
          {stepInfo[state.step].buttonText}
        </Button>
      )}
```

File: frontend/src/components/Button/index.js [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
-    <button onClick={onClick} className={styles.button}>
+    <button onClick={onClick} className={buttonClassName}>
       {children}
     </button>
   );
 };
 
+Button.defaultProps = {
+  isDisabled: false,
+  href: "",
```
Comment (mirhamasala) [inline frontend/src/components/Button/index.js:24]: Why set this to an empty string?

File: frontend/src/pages/SignupPage/index.js [hunk 3]
```diff
@@ -158,27 +158,22 @@ const SignupPage = () => {
...
     },
     STEP_4: {
+      buttonText: "I agree",
       component: (
         <PolicyForm handleChange={handleChange} submitData={submitData} />
       ),
     },
   };
 
+  let buttonHandler = state.step === STEP_4 ? submitData : continueToNextStep;
```
Comment (mirhamasala) [inline frontend/src/pages/SignupPage/index.js:168]: Why not `const`?

Comment (beasrodrigues24) [reply frontend/src/pages/SignupPage/index.js:168]: Oh right, João explained me better the difference between the two. I'll change it!

**Review by mirhamasala — APPROVED** [review]
✨❤️ Awesome. LGTM! 🚀

**Pull request merged**